src/gradualizer_highlight.erl

```diff
@@ -171,13 +171,13 @@ highlight_line(Line, N, {N, C1}, {N, C2}, Color) ->
     color_and_mark_line(Line, C1, C2, Color);
 highlight_line(Line, N, {L1, C1}, {L2, _C2}, Color) when L1 == N, N < L2 ->
     %% First line
-    color_and_mark_line(Line, C1, length(Line) + 1, Color);
+    color_and_mark_line(Line, C1, ?assert_type(length(Line) + 1, pos_integer()), Color);
```

Comment:
In the future, I think we could be able to handle the special case `non_neg_integer() + 1` = `pos_integer()`. It seems to appear in many places.

Reply:
Indeed, it pops up surprisingly frequently, so it would be nice to have it taken care of without extra assertions.

Reply:
Yes. We already have something for `pos_integer() - 1` <=> `non_neg_integer()` I think, but I'm not sure it works in both directions, i.e. (1) types pushed from result to operands and (2) result type inferred from the operands.